systems-management/nisysmgmt.yml

```diff
@@ -0,0 +1,2092 @@
+      summary: Get discovered systems summary
+      description: Get the number of network discovered systems. This number refers to the count of systems that are discovered and not managed.
+      responses:
+        200:
+          description: OK.
+          schema:
+            $ref: '#/definitions/DiscoveredSystemsSummary'
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/get-pending-systems-summary:
+    get:
+      operationId: GetPendingSystemsSummary
+      x-ni-privilege: Read
+      x-ni-operation: getSystems
+      tags: [systems]
+      summary: Get pending systems summary
+      description: Get the number of systems that are currently pending to be added to the master but have not yet been approved by the master.
+      responses:
+        200:
+          description: OK.
+          schema:
+            $ref: '#/definitions/PendingSystemsSummary'
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/generate-systems-report:
+    post:
+      operationId: GetSystemsReport
+      x-ni-privilege: Read
+      x-ni-operation: getSystems
+      tags: [systems]
+      summary: Get systems report
+      description: Get a report containing software/hardware information about the requested systems.
+      parameters:
+        - in: body
+          name: SystemsReportRequest
+          description: The type of the request followed by the list of system Ids.
+          required: true
+          schema:
+            $ref: '#/definitions/SystemsReportRequest'
+      produces:
+        - text/csv
+        - application/json
+      responses:
+        200:
+          description: OK.
+          schema:
+            description: The file data
+            type: file
+          examples:
+            application/octet-stream:
+              File data
+        400:
+          $ref: '#/responses/BadRequest'
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/create-groups:
+    post:
+      operationId: CreateGroups
+      x-ni-privilege: Write
+      x-ni-operation: createOrUpdateGroups
+      tags: [groups]
+      summary: Create groups
+      description: Create groups.
+      parameters:
+        - in: body
+          name: CreateGroupRequest
+          description: Request for creating groups.
+          required: true
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/CreateGroupRequest'
+      responses:
+        200:
+          description: OK.
+          schema:
+            $ref: '#/definitions/CreateGroupsPartialSuccessResponse'
+        204:
+          description: No Content.
+        400:
+          $ref: '#/responses/BadRequest'
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/groups:
+    get:
+      oper
```

Comment:
It seems this would need a 204 No Content response if no group matches the provided id